packages/@aws-cdk/aws-codepipeline/lib/pipeline.ts

```diff
@@ -600,7 +600,7 @@ export class Pipeline extends PipelineBase {
     // (otherwise we can't configure cross-account trust policies)
     if (action.isCrossAccount) {
       const artifactBucket = this.ensureReplicationResourcesExistFor(action).artifactBucket;
-      if (!artifactBucket.encryptionKey) {
+      if (artifactBucket.encryption != BucketEncryption.KMS && artifactBucket.encryption != BucketEncryption.KMS_MANAGED) {
```

Comment:
1. `artifactBucket` is of type `IBucket`. That interface does not have an `encryption` property.
2. The `!=` should be `!==`.